options/options.go

```diff
@@ -596,6 +596,20 @@ func (opts *ToolOptions) setOptionsFromURI(cs connstring.ConnString) error {
 					return ConflictingArgsErrorFormat("host", strings.Join(cs.Hosts, ","), opts.Host, "--host")
 				}
 			}
+		} else if len(cs.Hosts) > 0 {
+			hostPort := strings.Split(cs.Hosts[0], ":")
+			opts.Host = hostPort[0]
+
+			// a port might not be specified, e.g. `mongostat --discover`
+			if len(hostPort) == 2 {
+				if opts.Port != "" {
+					if hostPort[1] != opts.Port {
```

Comment:
Even we just need one host from the replica-set, but if you pick a host:Port randomly from the hosts list, we cannot guarantee the port number is what user specified in the startup cmd.

Reply:
If `--port` is set, I could check all of the hosts and pick the first one with a matching port?

Reply:
Um, that sounds OK to me. @tfogo do you have any suggestions?

Reply:
We need to loop through all elements of `cs.Hosts` to create `opts.Host` so we can check all the ports.